### ISSUE TITLE
Strip `iss` from incoming redirect URL callback in Node.js version

### ISSUE THREAD

**Issue by joachimvh** [inrupt/solid-client-authn-js#2985]
### Impacted package

Which packages do you think might be impacted by the bug ?

- [ ] solid-client-authn-browser
- [x] solid-client-authn-node
- [ ] solid-client-authn-core
- [ ] oidc-client-ext
- [ ] Other (please specify): ...

### Bug description

This is the same issue as https://github.com/inrupt/solid-client-authn-js/issues/2103, which was fixed in https://github.com/inrupt/solid-client-authn-js/pull/2584. But the fix only solved this for the browser version of the package. The Node.js version of the source runs into the same problem.

I ran into this when upgrading the `oidc-provider` library for CSS to the latest version. The internal tests started failing as those use the Node.js version of the library.

**Comment by jeswr** [comment]
Thanks for reporting @joachimvh - we will look into this!

**Comment by NSeydoux** [comment]
The browser module manages the URL the browser is redirected to though the browser APIs, which is why we can clean up the OpenID query params from the redirect URL. This isn't the case in the Node library, where the URL management is left to the Web framework used by the app. Restoring a previous session is implemented differently in Node, since there is a secure storage where the refresh token can be stored. Therefore, there is no front-channel redirect relying on a redirect URL, which was causing the issue in https://github.com/inrupt/solid-client-authn-js/issues/2103.

In other words, the application code must have already been managing the redirect URL and cleaning up the OpenID parameters, so maybe this is out of scope for `solid-client-authn-node` ?

Could you elaborate a bit on steps to reproduce the issue, so that I understand if I'm on the right track or completely mistaken?

**Comment by joachimvh** [comment]
The problem occurs since updating the `oidc-provider` library to v8. This is done on the branch in this PR, so you can use that to test it out: https://github.com/CommunitySolidServer/CommunitySolidServer/pull/1662

I've just tested it out, the problem can be reproduced by the `demoClientApp` in `packages/node/examples/demoClientApp`.

It might not be the exact same situation as the linked issue, but it seemed similar so I assumed it was the same. The problem occurs when creating a session and logging in. The server will return a redirect URL that then needs to be passed to `session.handleIncomingRedirect`. With the new version of the library, this redirect URL will have an `iss` query parameter, which will be passed along in the call here: https://github.com/inrupt/solid-client-authn-js/blob/c4a26cf60897f8e18f1b49fa79ad73ed823b6949/packages/node/src/login/oidc/incomingRedirectHandler/AuthCodeRedirectHandler.ts#L135

Which will then cause the server to throw an error as the redirect URL will not match because of this.

**Comment by elf-pavlik** [comment]
Sorry for last minute notice, in 20 min we plan to start a meeting which includes discussing adopting [RFC9207](https://www.rfc-editor.org/rfc/rfc9207.html) in Solid-OIDC https://github.com/solid/solid-oidc/issues/221

Meeting links in one of the last comments in that issue. Once again apologies for very late notice, still maybe someone is interested and able to join :crossed_fingers:

**Comment by NSeydoux** [comment]
@joachimvh I'm sorry for the delay, I realized why this was an issue, I'll fix this now.

**Comment by NSeydoux** [comment]
https://github.com/inrupt/solid-client-authn-js/releases/tag/1.17.4 should fix this issue.
